### PR TITLE
Update for NXOS - Add Fex Support

### PR DESCRIPTION
**Pull request by BobbyGR**
Adding in support for fex gathering. 

get_fex_information
get_fex_status

https://gist.github.com/BobbyGR/3766b884e9c9c6b6ab6a7f2be75ef71b
https://gist.github.com/BobbyGR/ed9247c35175bcdf632228b8a094ae83

### CODE REVIEW

**Comment by BobbyGR** [comment]
updated space

File: napalm_nxos/nxos.py [hunk 1]
```diff
@@ -204,7 +204,59 @@ def get_facts(self):
                 iface_list.append(intf)
 
         return results
+  
+    def get_fex_information(self):
+        item_list = list()
+
+        fex_command = 'show fex'
```
Comment (mirceaulinic) [inline napalm_nxos/nxos.py:211]: Firstly: what is "fex"? Is it specific to Nexus?

File: napalm_nxos/nxos.py [hunk 2]
```diff
@@ -204,7 +204,59 @@ def get_facts(self):
...
+
+        fex_raw_output = self.cli([fex_command]).get(fex_command, '')
```
Comment (mirceaulinic) [inline napalm_nxos/nxos.py:213]: This command does not work on my switches (nexus 5k):

``` bash
sw01.sfo04# show fex
                    ^
% Incomplete command at '^' marker.
```

Does `show fex drops` provide the same output?

File: napalm_nxos/nxos.py [hunk 3]
```diff
@@ -204,7 +204,59 @@ def get_facts(self):
...
+             i = i + 1
+           except:
+             i = 100
 
+        return results
+        
+    def get_fex_status(self):
+        item_list = list()
+
+        fex_command = 'show fex detail'
```
Comment (mirceaulinic) [inline napalm_nxos/nxos.py:238]: Again, this command does not exist on 5k:

```
sw01.sfo04# show fex detail
                      ^
% Invalid command at '^' marker.
```

**Review by mirceaulinic — CHANGES_REQUESTED** [review]
I missed this PR previously.

A couple of things:
- is this fex something specific to nexus? Is there equiv on other operating systems?
- whenever you propose a new feature, we need tests written, e.g.: https://github.com/napalm-automation/napalm-base/pull/69 You can open a PR and we'll help you to sort it out
- fex-commands seems to have JSON-equiv. Please double check, this would be the preferred way rather than scraping the CLI output.

Let me know if you have any questions.

File: napalm_nxos/nxos.py (hunk 1 above)
Comment (BobbyGR) [reply napalm_nxos/nxos.py:211]: Fabric Extenders for Nexus 5k/2k.
DS# sh fex
  FEX         FEX           FEX              FEX              Fex

## Number    Description      State            Model            Serial

102      ATL1-AS3P                Online   N2K-C2248TP-E-1GE   SSIxxxxxxxx
etc.
etc
etc.

File: napalm_nxos/nxos.py (hunk 2 above)
Comment (BobbyGR) [reply napalm_nxos/nxos.py:213]: Please ensure that 'Feature Fex' is enabled and you are using fex's

File: napalm_nxos/nxos.py (hunk 3 above)
Comment (BobbyGR) [reply napalm_nxos/nxos.py:238]: DS# sh fex detail
FEX: 102 Description: AS3P   state: Online
  FEX version: 7.2(1)N1(1) [Switch version: 7.2(1)N1(1)]
  FEX Interim version: 7.2(1)N1(1)
  Switch Interim version: 7.2(1)N1(1)
  Extender Serial: SSIXXXXXXX
  Extender Model: N2K-C2248TP-E-1GE,  Part No: 73-13671-01
  Card Id: 149, Mac Addr: 34:a8:XXXXXXXX, Num Macs: 64
  Module Sw Gen: 12594  [Switch Sw Gen: 21]
  post level: complete
 pinning-mode: static    Max-links: 1
  Fabric port for control traffic: Eth1/31
  FCoE Admin: false
  FCoE Oper: true
  FCoE FEX AA Configured: false
  Fabric interface state:
    Po102 - Interface Up. State: Active
    Eth1/31 - Interface Up. State: Active
    Eth2/15 - Interface Up. State: Active
  Fex Port        State  Fabric Port
       Eth102/1/1    Up       Po102
       Eth102/1/2    Up       Po102
       Eth102/1/3    Up       Po102
       Eth102/1/4    Up       Po102
       Eth102/1/5  Down       Po102
       Eth102/1/6  Down       Po102
       Eth102/1/7    Up       Po102
       Eth102/1/8    Up       Po102
       Eth102/1/9    Up       Po102
      Eth102/1/10    Up       Po102
      Eth102/1/11    Up       Po102

**Comment by BobbyGR** [comment]
Updated to show fex output if you are using fex's

**Comment by BobbyGR** [comment]
"fex-commands seems to have JSON-equiv. Please double check, this would be the preferred way rather than scraping the CLI output."

Going off of other show commands in the NX-OS family in napalm.  can be for a future version?  This version works now, and is in "production".

**Comment by mirceaulinic** [comment]
Hey @BobbyGR - Thanks for clarifying this!

I don't want to disappoint you but this pull request might not be suitable under napalm as we are trying to keep the library consistent across network operating systems. As this feature by definition is specific to Nexus switches only, I think the right place to open this PR is pycsco.
However I appreciate the work you've done here and looks like interesting data to have - could you think of similar functions for other vendors and introduce this as something like `get_vendor_extenders` etc.?

**Comment by BobbyGR** [comment]
Will find another home for this stuff

**Pull request closed without merging**